File: AsyncTask/app/src/main/java/com/example/asynctask/MainActivity.java
package com.example.asynctask;

import androidx.appcompat.app.AppCompatActivity;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button btnStart,btnStop;
    TextView tvContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnStart = findViewById(R.id.btnStart);
        btnStop = findViewById(R.id.btnStop);
        tvContent = findViewById(R.id.tvContent);


        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                CountTask countTask = new CountTask();
                countTask.execute(100000L);
//                /*This is non-UI thread*/
//                Thread thread = new Thread(new Runnable() {
//                    @Override
//                    public void run() {
//                        try {
//                            Thread.sleep(5000);
//                        } catch (InterruptedException e) {
//                            e.printStackTrace();
//                        }
//
//                        runOnUiThread(new Runnable() {
//                            @Override
//                            public void run() {
//                                tvContent.setText("Counting Done!");
//                            }
//                        });
//                    }
//                });
//                thread.start();
            }
        });

        btnStop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this,
                        "Timer Stopped ",
                        Toast.LENGTH_SHORT).show();
            }
        });

    }

    class CountTask extends AsyncTask<Long,Double,String>{

        /*Runs on Main Thread*/
        @Override
        protected void onPreExecute() {
            super.onPreExecute();
            tvContent.setText("Counting Started");
        }

        /*Runs on Background Thread*/
        @Override
        protected String doInBackground(Long... longs) {

            Long countThreshold = longs[0];/*If there is only one "Long Variable" present*/
//            try {
//                Thread.sleep(countThreshold);
//            } catch (InterruptedException e) {
//                e.printStackTrace();
//            }
            /*or*/
            for (int i = 0; i < countThreshold ; i++){
                /*Here we can also show How onProgress method works*/
                if(i%10000 == 0){
                    publishProgress((double) i);
                }
            }
//            /*For more than one long variable*/
//            for (int i = 0; i < longs.length ; i++){
//                long currentCount = longs[i];
//            }
            return "Counting Till " + countThreshold + " done";
        }

        /*Runs on Main Thread*/
        @Override
        protected void onProgressUpdate(Double... values) {
            super.onProgressUpdate(values);

            Double currentProgress = values[0];
            Toast.makeText(MainActivity.this,
                    "Count is : " + currentProgress,
                    Toast.LENGTH_SHORT).show();
        }

        /*Runs on Main Thread*/
        @Override
        protected void onPostExecute(String s) {
            super.onPostExecute(s);
            tvContent.setText(s);
        }

    }

}

File: Kotlin/app/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Adapter
import android.widget.Toast
import androidx.core.content.ContextCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    // can be reassigned
    var count : Int = 0
    // can't be reassigned
    val valCount : Int = 0

    lateinit var noteAdapter: NoteAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        noteAdapter = NoteAdapter()

        count++
        count--

        
        tvText1.setOnClickListener{
            tvText2.setTextColor(ContextCompat.getColor(this,R.color.colorAccent))
        }

//        valCount++
    }

    fun increment(){
        count++
    }

    fun decrement(){
        count--
    }

    fun countSquare() : Int{
        return count*count
    }
    /*OR*/
    fun singleLineFunCountSquare() = count*count


    fun showToast(number : Int){
        showShortToast(this,"Short Toast")
        showLongToast(this,"Long Toast")
        Toast.makeText(this,"the number is $number" , Toast.LENGTH_SHORT).show()
    }

    fun printEvenOrOdd(number: Int){

        val isEven = if (number % 2 == 0) true else false

        Toast.makeText(this,
                "$number is : ${ if (number % 2 == 0 ) "even" else "odd" }"
                ,Toast.LENGTH_SHORT).show()

    }


}
